pkg/helm/actions/auth.go

```diff
@@ -0,0 +1,125 @@
+package actions
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/api/helm/v1beta1"
+	"helm.sh/helm/v3/pkg/action"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+func setUpAuthentication(chartPathOptions *action.ChartPathOptions, connectionConfig *v1beta1.ConnectionConfig, coreClient corev1client.CoreV1Interface) ([]*os.File, error) {
+	tlsFiles := []*os.File{}
+	var secretNamespace, configMapName, configMapNameSpace, secretName string
+	//set up tls cert and key
+	if connectionConfig.TLSClientConfig != (configv1.SecretNameReference{}) {
+		chartPathOptions.RepoURL = connectionConfig.URL
+		secretName = connectionConfig.TLSClientConfig.Name
+		secretNamespace = configNamespace
+		secret, err := coreClient.Secrets(secretNamespace).Get(context.TODO(), secretName, v1.GetOptions{})
+		if err != nil {
+			return nil, fmt.Errorf("Failed to GET secret %s from %vreason %v", secretName, secretNamespace, err)
+		}
+		tlsCertBytes, found := secret.Data[tlsSecretCertKey]
+		if !found {
+			return nil, fmt.Errorf("Failed to find %s key in secret %s", tlsSecretCertKey, secretName)
```

Comment:
I would suggest using `%q` instead of `%s` for names/strings since its more readable from the logs.

Reply:
Here and across the whole PR, since both of them are being used.

Reply:
sure